week3/homework/src/myFunctions.js

```diff
@@ -77,4 +88,29 @@ function markAsDoneOrMarkAsNotDone(request, response, trueOrFalse) {
   });
 }
 
-module.exports = { createToDo, readToDo, deleteToDo, clearToDos, markAsDoneOrMarkAsNotDone };
+function updateToDo(request, response) {
+  fs.readFile('./todos.json', 'utf8', (error, contents) => {
+    if (error) {
+      response.status(500).send('Whats wrong with you');
+    } else {
+      const updatedContent = JSON.parse(contents);
+      updatedContent[request.params.id - 1].todo = request.body.todo;
+      const updatedNewContent = JSON.stringify(updatedContent);
+      fs.writeFile('./todos.json', updatedNewContent, err => {
```

Comment:
stringify followed by writeFile is a a perfect candidate for applying a refactoring technique called extract as function which helps with duplicate code.
```
function saveTodos(todos) {
    const todosString= JSON.stringify(todos);
    fs.writeFile('./todos.json', todosString, err => {
      if (err) throw err;
    });
}
```

same goes for `readFile` and `JSON.parse`